lib/lightning_web/channels/attempt_channel.ex

```diff
@@ -90,6 +90,7 @@ defmodule LightningWeb.AttemptChannel do
     %{attempt: attempt, scrubber: scrubber} = socket.assigns
 
     with credential <- Attempts.get_credential(attempt, id) || :not_found,
+         credential <- Credentials.cast_body(credential),
```

Comment:
I'm not sure this is necessary. If the credentials are persisted correctly we shouldn't cast them a second time. 

If we absolutely have to cast them again, I really think we shouldn't be replacing the `:body` key on the credential. I would prefer the cast body was the return value directly, and we send that out straight to the worker.

Originally I wanted these to be cast automatically, but it was more trouble than it was worth (at the time). I want to avoid situations where a model is changed and not persisted; we run the risk of downstream functions assuming everything is a certain shape and not getting the opportunity to know it was changed (or passed through `cast_body`).

Again I would prefer we not re-cast these, I don't believe this is the place for this code to have any say as to the validity or shape of something. All it fixes in my mind at this precise moment is previously (incorrectly) saved credentials appearing as if they are correct.

Reply:
Hey Stu, this is for existing credentials. I wanted to avoid touching/changing them persistently but I am working on the migration ...